src/LocoEvent.php

```diff
@@ -97,7 +97,7 @@ public function hasArgument($key) {
    *
    * @throws \InvalidArgumentException if key does not exist in $this->args
    */
-  public function &offsetGet($key) {
+  public function &offsetGet($key): mixed {
```

Comment:
Doesn't 'mixed' require php8? Ie break compat with php7?

https://wiki.php.net/rfc/mixed_type_v2

Reply:
I think it is ignored by php7 I think I was able to run unit tests on my local php7.4 instance but might be worth double checking

Reply:
OK,I pulled it up on `bknix-min` / `php72`:

```
[bknix-min:~/bknix] ~/src/loco/bin/loco run
PHP Fatal error:  Uncaught TypeError: Return value of Loco\LocoEvent::offsetGet() must be an instance of Loco\mixed, string returned in /home/totten/src/loco/src/LocoEvent.php:101
Stack trace:
#0 /home/totten/bknix/.loco/plugin/bknix-overrides.php(63): Loco\LocoEvent->offsetGet('file')
#1 /home/totten/src/loco/vendor/symfony/event-dispatcher/EventDispatcher.php(264): Loco\LocoPlugins->BknixOverrides\{closure}(Object(Loco\LocoEvent), 'loco.config.fil...', Object(Symfony\Component\EventDispatcher\EventDispatcher))
#2 /home/totten/src/loco/vendor/symfony/event-dispatcher/EventDispatcher.php(239): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'loco.config.fil...', Object(Loco\LocoEvent))
#3 /home/totten/src/loco/vendor/symfony/event-dispatcher/EventDispatcher.php(73): Symfony\Component\EventDispatcher\EventDispatcher->callListeners(Array, 'loco.config.fil...', Object(Loco\LocoEvent))
#4 /home/totten/src/loco/src/Loco.php(71): Symfony\Component\EventDispatcher\EventDispatcher->dispatch(Object(Loco\LocoEvent), in /home/totten/src/loco/src/LocoEvent.php on line 101

Fatal error: Uncaught TypeError: Return value of Loco\LocoEvent::offsetGet() must be an instance of Loco\mixed, string returned in /home/totten/src/loco/src/LocoEvent.php on line 101

TypeError: Return value of Loco\LocoEvent::offsetGet() must be an instance of Loco\mixed, string returned in /home/totten/src/loco/src/LocoEvent.php on line 101

Call Stack:
    0.0004     419192   1. {main}() /home/totten/src/loco/bin/loco:0
    0.0026     823056   2. Loco\Application::main() /home/totten/src/loco/bin/loco:34
    0.0079    1681344   3. Loco\Application->run() /home/totten/src/loco/src/Application.php:25
    0.0134    2027680   4. Loco\Application->doRun() /home/totten/src/loco/vendor/symfony/console/Application.php:149
    0.0134    2027680   5. Loco\Application->doRun() /home/totten/src/loco/src/Application.php:58
    0.0136    2031072   6. Loco\Application->doRunCommand() /home/totten/src/loco/vendor/symfony/console/Application.php:273
    0.0137    2031072   7. Loco\Command\RunCommand->run() /home/totten/src/loco/vendor/symfony/console/Application.php:1009
    0.0138    2032736   8. Loco\Command\RunCommand->execute() /home/totten/src/loco/vendor/symfony/console/Command/Command.php:255
    0.0138    2032736   9. Loco\Command\RunCommand->executeInManagedMode() /home/totten/src/loco/src/Command/RunCommand.php:42
    0.0138    2032736  10. Loco\Command\RunCommand->initSystem() /home/totten/src/loco/src/Command/RunCommand.php:93
    0.0267    2556464  11. Loco\Loco::filter() /home/totten/src/loco/src/Command/LocoCommandTrait.php:45
    0.0267    2556544  12. Symfony\Component\EventDispatcher\EventDispatcher->dispatch() /home/totten/src/loco/src/Loco.php:71
    0.0267    2557336  13. Symfony\Component\EventDispatcher\EventDispatcher->callListeners() /home/totten/src/loco/vendor/symfony/event-dispatcher/EventDispatcher.php:73
    0.0267    2557336  14. Symfony\Component\EventDispatcher\EventDispatcher->doDispatch() /home/totten/src/loco/vendor/symfony/event-dispatcher/EventDispatcher.php:239
    0.0267    2557336  15. Loco\LocoPlugins->BknixOverrides\{closure:/home/totten/bknix/.loco/plugin/bknix-overrides.php:61-76}() /home/totten/src/loco/vendor/symfony/event-dispatcher/EventDispatcher.php:264
    0.0267    2557336  16. Loco\LocoEvent->offsetGet() /home/totten/bknix/.loco/plugin/bknix-overrides.php:63
```

But fortunately the `void` seems OK.

How important is the `mixed` part?

Reply:
@totten it breaks on php8.1 because ArrayAccess I think it is is type hinted at that version

Reply:
OK, well, that's yucky then.

Pushed an update so that older PHP 7.x and newer PHP 8.1 can use different signatures for `offsetGet()`.